src/main/java/net/snowflake/ingest/utils/SubColumnFinder.java

```diff
@@ -52,6 +54,13 @@ public String getDotPath(String id) {
     return accessMap.get(id).dotPath;
   }
 
+  public List<String> getPath(String id) {
```

Comment:
you're returning internal state and the callsite can mutate this list accidentally.
Please return an UnmodifiableList from this API.

Reply:
Made it to `UnmodifiableList()`